Description: Set compilation level to -02 on mips (it's -03 by default upstream)
 .
 Optimizing further drives gcc nuts and produce ICE.
 .
 https://buildd.debian.org/status/fetch.php?pkg=widelands&arch=mips&ver=1%3A18-1&stamp=1393243179
Author: Martin Quinson
Forwarded-Upstream: should be

---
 CMakeLists.txt |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

Index: b/CMakeLists.txt
===================================================================
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -264,10 +264,12 @@
     OUTPUT_VARIABLE WLBUILD_COMPILERVERSION
   )
   STRING(REGEX REPLACE ".*(4)\\.(5)\\.([0-9]).*" "\\1.\\2.\\3" WLBUILD_COMPILERVERSION_REP ${WLBUILD_COMPILERVERSION})
-  IF (WLBUILD_COMPILERVERSION_REP STREQUAL "4.5.1" OR WLBUILD_COMPILERVERSION_REP STREQUAL "4.5.2")
-    message("Detected gcc ${WLBUILD_COMPILERVERSION_REP}")
+  IF (WLBUILD_COMPILERVERSION_REP STREQUAL "4.5.1" OR WLBUILD_COMPILERVERSION_REP STREQUAL "4.5.2" OR PROCESSOR_ARCHITECTURE STREQUAL "mips64" OR PROCESSOR_ARCHITECTURE STREQUAL "mips")
+    message("Detected gcc ${WLBUILD_COMPILERVERSION_REP} on ${PROCESSOR_ARCHITECTURE}")
     message("Suffering from gcc bug, disabling -O3")
     set (WL_COMPILERFLAG_OPTIMIZATIONS "-O2")
+  else (WLBUILD_COMPILERVERSION_REP STREQUAL "4.5.1" OR WLBUILD_COMPILERVERSION_REP STREQUAL "4.5.2" OR PROCESSOR_ARCHITECTURE STREQUAL "mips64" OR PROCESSOR_ARCHITECTURE STREQUAL "mips")
+    message("Detected architecture: ${PROCESSOR_ARCHITECTURE}")
   ENDIF (WLBUILD_COMPILERVERSION_REP STREQUAL "4.5.1" OR WLBUILD_COMPILERVERSION_REP STREQUAL "4.5.2")
 ENDIF (CMAKE_COMPILER_IS_GNUCXX)
 
